### PR TITLE
Add support for setting the pastebin line limit in room state

### LINKED ISSUE #1293: Fix 4 linter warnings
As per #1289 

```
/home/jaller94/Git/matrix-appservice-irc/src/datastore/postgres/PgDataStore.ts
  133:43  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  269:9   warning  'allowUnset' is defined but never used    @typescript-eslint/no-unused-vars
  499:49  warning  Forbidden non-null assertion              @typescript-eslint/no-non-null-assertion
  505:35  warning  Forbidden non-null assertion              @typescript-eslint/no-non-null-assertion
```

### PR DESCRIPTION
**Pull request by Half-Shot**
This feature allows room admins to adjust the amount of lines required to end up in a pastebin by modifying the room state. This PR also paves the way for other features (like reply formats).

Fixed #1293  (accidentally)

### CODE REVIEW

File: src/bridge/MatrixHandler.ts [hunk 1]
```diff
@@ -1031,7 +1031,8 @@ export class MatrixHandler {
 
         // Generate an array of individual messages that would be sent
         const potentialMessages = ircClient.getSplitMessages(ircRoom.channel, ircAction.text);
-        const lineLimit = ircRoom.server.getLineLimit();
+        const roomLineLimit = await this.ircBridge.roomConfigs.getLineLimit(event.room_id, ircRoom);
```
Comment (Half-Shot) [inline src/bridge/MatrixHandler.ts:1034]: I'm a little worried about the delay in handling messages if room state is too slow, but hopefully the query will be subsecond quick.

Comment (Half-Shot) [reply src/bridge/MatrixHandler.ts:1034]: I wonder if this just needs a timeout.

Comment (Half-Shot) [reply src/bridge/MatrixHandler.ts:1034]: I added a timeout.

File: src/datastore/postgres/PgDataStore.ts [hunk 2]
```diff
@@ -16,7 +16,13 @@ limitations under the License.
```
Comment (Half-Shot) [inline src/datastore/postgres/PgDataStore.ts:16]: I made these changes to fix the lint, probably could have done with it's own PR but oh well.

File: src/bridge/RoomConfig.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            log.debug(
+                `Stored new config for ${cacheKey}: ${keyedConfig ? 'No config set' : JSON.stringify(keyedConfig)}`
+            );
+            this.cache.set(cacheKey, keyedConfig || undefined);
+            return keyedConfig as RoomConfigContent|null;
+        }
+        // We don't want to spend too long trying to fetch the state, so return null.
+        return Promise.race([
+            internalFunc(),
+            new Promise<null>(res => setTimeout(() => res(null), STATE_TIMEOUT_MS)),
```
Comment (jaller94) [inline src/bridge/RoomConfig.ts:59]: Already tested with the TS linter.
```suggestion
            new Promise<null>(res => setTimeout(res, STATE_TIMEOUT_MS)),
```

File: src/bridge/RoomConfig.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+     * @param ircRoom The IRC roomId. Optional.
+     * @returns The number of lines required for a pastebin. `null` means no limit set in the room.
+     */
+    public async getLineLimit(roomId: string, ircRoom?: IrcRoom) {
+        const roomState = await this.getRoomState(roomId, ircRoom);
+        if (typeof roomState?.lineLimit !== 'number' || roomState.lineLimit <= 0) {
+            // A missing line limit or an invalid one is considered invalid.
+            return null;
+        }
+        return Math.min(roomState.lineLimit, this.config?.lineLimitMax || roomState.lineLimit);
```
Comment (jaller94) [inline src/bridge/RoomConfig.ts:90]: Given the validation above, this make no technical difference, but would make the intention more clear.
```suggestion
        return Math.min(roomState.lineLimit, this.config?.lineLimitMax ?? roomState.lineLimit);
```

**Review by jaller94 — APPROVED** [review]
Looks good.

File: docs/room_configuration.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+- You should be able to click Edit to edit the content, and then hit Send to adjust the config.
+
+If an event does not exist yet, you can instead do:
+
+- Typing `/devtools`.
+- Click Send Custom Event.
+- Click the Event button to change the type to a **State Event**.
+- The event type must be `org.matrix.appservice-irc.config`
+- The state key can be left blank.
+- Entering the `Event Content` as described below.
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply docs/room_configuration.md:28]: Very true, thanks for making we aware :+1:
